### PR TITLE
Adding logstash, prometheus, privacyidea RT env

### PR DESCRIPTION
**Pull request by sand-hya**
logstash: https://github.com/Be-Secure/Be-Secure/issues/235
prometheus: https://github.com/Be-Secure/Be-Secure/issues/299
privacyidea: https://github.com/Be-Secure/Be-Secure/issues/377

### CODE REVIEW

File: logstash/0.0.1/besman-logstash-RT-env-config.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+BESMAN_ARTIFACT_TRIGGER_PLAYBOOK_PATH: $BESMAN_DIR/tmp/$BESMAN_ARTIFACT_NAME #***
+
+#BESMAN_ARTIFACT_TRIGGER_PLAYBOOK - Name of the trigger playbook which runs the ansible roles.
+BESMAN_ARTIFACT_TRIGGER_PLAYBOOK: besman-$BESMAN_ARTIFACT_NAME-RT-trigger-playbook.yaml #***
+
+# BESMAN_DISPLAY_SKIPPED_ANSIBLE_HOSTS - If the users likes to display all the skipped steps, set it to true.
+# Default value is false
+BESMAN_DISPLAY_SKIPPED_ANSIBLE_HOSTS: false #***
+
+ASSESSMENT_STEP:
```
Comment (asa1997) [inline logstash/0.0.1/besman-logstash-RT-env-config.yaml:65]: You can remove this.

File: logstash/0.0.1/besman-logstash-RT-env.sh [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    # Function to validate ruby installation
+    validate_ruby() {
+        if ! command -v ruby &>/dev/null; then
+            echo "ruby is not installed."
+            return 1
+        fi
+    }
+
+    # Main validation function
+    validate_environment() {
```
Comment (asa1997) [inline logstash/0.0.1/besman-logstash-RT-env.sh:103]: What is the use of this functions? Also, please provide this function definition outside other functions.

File: privacyidea/0.0.1/besman-privacyidea-RT-env-config.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# The variables that start with BESMAN_ are converted to environment vars.
+# If you wish to add any other vars that should be used globally, add the var using the below format.
+# BESMAN_<var name>: <value>
+# If you are not using any particular value, remove it or comment it(#).
+#*** - These variables should not be removed, nor left empty.
+# BESMAN_ORG - used to mention where you should clone the repo from, default value is Be-Secure
+BESMAN_ORG: Be-Secure #***
+
+# BESMAN_ARTIFACT_TYPE - project/ml model/training dataset
+BESMAN_ARTIFACT_TYPE: # project/ml model/training dataset #***
```
Comment (asa1997) [inline privacyidea/0.0.1/besman-privacyidea-RT-env-config.yaml:13]: missing parameter - `project`

File: privacyidea/0.0.1/besman-privacyidea-RT-env-config.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+# BESMAN_ARTIFACT_NAME - name of the artifact under assessment.
+BESMAN_ARTIFACT_NAME: privacyidea #***
+
+# BESMAN_ARTIFACT_VERSION - version of the artifact under assessment.
+BESMAN_ARTIFACT_VERSION: #Enter the version of the artifact here. #***
```
Comment (asa1997) [inline privacyidea/0.0.1/besman-privacyidea-RT-env-config.yaml:19]: missing parameter

File: privacyidea/0.0.1/besman-privacyidea-RT-env-config.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+BESMAN_ARTIFACT_TRIGGER_PLAYBOOK_PATH: $BESMAN_DIR/tmp/$BESMAN_ARTIFACT_NAME #***
+
+#BESMAN_ARTIFACT_TRIGGER_PLAYBOOK - Name of the trigger playbook which runs the ansible roles.
+BESMAN_ARTIFACT_TRIGGER_PLAYBOOK: besman-$BESMAN_ARTIFACT_NAME-privacyidea-trigger-playbook.yaml #***
+
+# BESMAN_DISPLAY_SKIPPED_ANSIBLE_HOSTS - If the users likes to display all the skipped steps, set it to true.
+# Default value is false
+BESMAN_DISPLAY_SKIPPED_ANSIBLE_HOSTS: false #***
+
+ASSESSMENT_STEP:
```
Comment (asa1997) [inline privacyidea/0.0.1/besman-privacyidea-RT-env-config.yaml:65]: please remove this.

File: privacyidea/0.0.1/besman-privacyidea-RT-env.sh [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+}
+
+function __besman_validate
+{
+    __besman_check_for_trigger_playbook "$BESMAN_ARTIFACT_TRIGGER_PLAYBOOK_PATH/$BESMAN_ARTIFACT_TRIGGER_PLAYBOOK"
+    [[ "$?" -eq 1 ]] && __besman_create_ansible_playbook
+    __besman_run_ansible_playbook_extra_vars "$BESMAN_ARTIFACT_TRIGGER_PLAYBOOK_PATH/$BESMAN_ARTIFACT_TRIGGER_PLAYBOOK" "bes_command=validate role_path=$BESMAN_ANSIBLE_ROLES_PATH" || return 1
+    # Please add the rest of the code here for validate
+    validate_python() {
```
Comment (asa1997) [inline privacyidea/0.0.1/besman-privacyidea-RT-env.sh:90]: Please move this func definition outside

File: prometheus/0.0.1/besman-prometheus-RT-env-config.yaml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+# The variables that start with BESMAN_ are converted to environment vars.
+# If you wish to add any other vars that should be used globally, add the var using the below format.
+# BESMAN_<var name>: <value>
+# If you are not using any particular value, remove it or comment it(#).
+#*** - These variables should not be removed, nor left empty.
+# BESMAN_ORG - used to mention where you should clone the repo from, default value is Be-Secure
+BESMAN_ORG: Be-Secure #***
+
+# BESMAN_ARTIFACT_TYPE - project/ml model/training dataset
+BESMAN_ARTIFACT_TYPE: # project/ml model/training dataset #***
```
Comment (asa1997) [inline prometheus/0.0.1/besman-prometheus-RT-env-config.yaml:13]: missing parameter

File: prometheus/0.0.1/besman-prometheus-RT-env-config.yaml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+# BESMAN_ARTIFACT_NAME - name of the artifact under assessment.
+BESMAN_ARTIFACT_NAME: prometheus #***
+
+# BESMAN_ARTIFACT_VERSION - version of the artifact under assessment.
+BESMAN_ARTIFACT_VERSION: #Enter the version of the artifact here. #***
```
Comment (asa1997) [inline prometheus/0.0.1/besman-prometheus-RT-env-config.yaml:19]: missing parameter

File: prometheus/0.0.1/besman-prometheus-RT-env-config.yaml [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+BESMAN_ARTIFACT_TRIGGER_PLAYBOOK_PATH: $BESMAN_DIR/tmp/$BESMAN_ARTIFACT_NAME #***
+
+#BESMAN_ARTIFACT_TRIGGER_PLAYBOOK - Name of the trigger playbook which runs the ansible roles.
+BESMAN_ARTIFACT_TRIGGER_PLAYBOOK: besman-$BESMAN_ARTIFACT_NAME-RT-trigger-playbook.yaml #***
+
+# BESMAN_DISPLAY_SKIPPED_ANSIBLE_HOSTS - If the users likes to display all the skipped steps, set it to true.
+# Default value is false
+BESMAN_DISPLAY_SKIPPED_ANSIBLE_HOSTS: false #***
+
+ASSESSMENT_STEP:
```
Comment (asa1997) [inline prometheus/0.0.1/besman-prometheus-RT-env-config.yaml:65]: please remove this.

File: prometheus/0.0.1/besman-prometheus-RT-env.sh [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+    __besman_run_ansible_playbook_extra_vars "$BESMAN_ARTIFACT_TRIGGER_PLAYBOOK_PATH/$BESMAN_ARTIFACT_TRIGGER_PLAYBOOK" "bes_command=remove role_path=$BESMAN_ANSIBLE_ROLES_PATH" || return 1
+    if [[ -d $BESMAN_ARTIFACT_DIR ]]; then
+        __besman_echo_white "Removing $BESMAN_ARTIFACT_DIR..."
+        rm -rf "$BESMAN_ARTIFACT_DIR"
+    else
+        __besman_echo_yellow "Could not find dir $BESMAN_ARTIFACT_DIR"
+    fi
+    # Please add the rest of the code here for uninstallation
+    # Function to uninstall go
+    uninstall_go() {
```
Comment (asa1997) [inline prometheus/0.0.1/besman-prometheus-RT-env.sh:74]: you can add the code in this function itself. If you want to separate the functions then move this outside other functions.

File: prometheus/0.0.1/besman-prometheus-RT-env.sh [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+    # Function to validate go installation
+    validate_go() {
+        if ! command -v go &>/dev/null; then
+            echo "go is not installed."
+            return 1
+        fi
+    }
+
+    # Main validation function
+    validate_environment() {
```
Comment (asa1997) [inline prometheus/0.0.1/besman-prometheus-RT-env.sh:107]: move this outside other function definitions.

File: logstash/0.0.1/besman-logstash-RT-env-config.yaml [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+# The variables that start with BESMAN_ are converted to environment vars.
+# If you wish to add any other vars that should be used globally, add the var using the below format.
+# BESMAN_<var name>: <value>
+# If you are not using any particular value, remove it or comment it(#).
+#*** - These variables should not be removed, nor left empty.
+# BESMAN_ORG - used to mention where you should clone the repo from, default value is Be-Secure
+BESMAN_ORG: Be-Secure #***
+
+# BESMAN_ARTIFACT_TYPE - project/ml model/training dataset
+BESMAN_ARTIFACT_TYPE: # project/ml model/training dataset #***
```
Comment (asa1997) [inline logstash/0.0.1/besman-logstash-RT-env-config.yaml:13]: missing parameter

File: logstash/0.0.1/besman-logstash-RT-env-config.yaml [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+
+# BESMAN_ARTIFACT_NAME - name of the artifact under assessment.
+BESMAN_ARTIFACT_NAME: logstash #***
+
+# BESMAN_ARTIFACT_VERSION - version of the artifact under assessment.
+BESMAN_ARTIFACT_VERSION: #Enter the version of the artifact here. #***
```
Comment (asa1997) [inline logstash/0.0.1/besman-logstash-RT-env-config.yaml:19]: missing parameter

**Review by asa1997 — CHANGES_REQUESTED** [review]
@sand-hya Please resolve the conflicts

**Comment by sand-hya** [comment]
Hi @asa1997, performed the changes requested and resolved the conflicts

**Review by asa1997 — CHANGES_REQUESTED** [review]
@sand-hya Please resolve the conflicts. Also please provide the names of the env on the title

**Pull request closed without merging**